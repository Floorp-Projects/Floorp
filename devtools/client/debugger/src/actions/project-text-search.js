/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

/**
 * Redux actions for the search state
 * @module actions/search
 */

import { isFulfilled } from "../utils/async-value";
import { findSourceMatches } from "../workers/search";
import {
  getSource,
  hasPrettySource,
  getSourceList,
  getSourceContent,
} from "../selectors";
import { isThirdParty } from "../utils/source";
import { loadSourceText } from "./sources/loadSourceText";
import {
  statusType,
  getTextSearchOperation,
  getTextSearchStatus,
} from "../reducers/project-text-search";

export function addSearchQuery(cx, query) {
  return { type: "ADD_QUERY", cx, query };
}

export function addOngoingSearch(cx, ongoingSearch) {
  return { type: "ADD_ONGOING_SEARCH", cx, ongoingSearch };
}

export function addSearchResult(cx, sourceId, filepath, matches) {
  return {
    type: "ADD_SEARCH_RESULT",
    cx,
    result: { sourceId, filepath, matches },
  };
}

export function clearSearchResults(cx) {
  return { type: "CLEAR_SEARCH_RESULTS", cx };
}

export function clearSearch(cx) {
  return { type: "CLEAR_SEARCH", cx };
}

export function updateSearchStatus(cx, status) {
  return { type: "UPDATE_STATUS", cx, status };
}

export function closeProjectSearch(cx) {
  return ({ dispatch, getState }) => {
    dispatch(stopOngoingSearch(cx));
    dispatch({ type: "CLOSE_PROJECT_SEARCH" });
  };
}

export function stopOngoingSearch(cx) {
  return ({ dispatch, getState }) => {
    const state = getState();
    const ongoingSearch = getTextSearchOperation(state);
    const status = getTextSearchStatus(state);
    if (ongoingSearch && status !== statusType.done) {
      ongoingSearch.cancel();
      dispatch(updateSearchStatus(cx, statusType.cancelled));
    }
  };
}

export function searchSources(cx, query) {
  let cancelled = false;

  const search = async ({ dispatch, getState }) => {
    dispatch(stopOngoingSearch(cx));
    await dispatch(addOngoingSearch(cx, search));
    await dispatch(clearSearchResults(cx));
    await dispatch(addSearchQuery(cx, query));
    dispatch(updateSearchStatus(cx, statusType.fetching));
    let validSources = getSourceList(getState()).filter(
      source => !hasPrettySource(getState(), source.id) && !isThirdParty(source)
    );
    // Sort original entries first so that search results are more useful.
    // See bug 1642778.
    validSources = [
      ...validSources.filter(x => x.isOriginal),
      ...validSources.filter(x => !x.isOriginal),
    ];
    for (const source of validSources) {
      if (cancelled) {
        return;
      }
      await dispatch(loadSourceText({ cx, source }));
      await dispatch(searchSource(cx, source.id, query));
    }
    dispatch(updateSearchStatus(cx, statusType.done));
  };

  search.cancel = () => {
    cancelled = true;
  };

  return search;
}

export function searchSource(cx, sourceId, query) {
  return async ({ dispatch, getState }) => {
    const source = getSource(getState(), sourceId);
    if (!source) {
      return;
    }

    const content = getSourceContent(getState(), source.id);
    let matches = [];
    if (content && isFulfilled(content) && content.value.type === "text") {
      matches = await findSourceMatches(source.id, content.value, query);
    }
    if (!matches.length) {
      return;
    }
    dispatch(addSearchResult(cx, source.id, source.url, matches));
  };
}
