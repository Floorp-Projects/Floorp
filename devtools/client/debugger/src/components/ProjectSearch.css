/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

.search-container {
  position: absolute;
  top: var(--editor-header-height);
  left: 0;
  width: calc(100% - 1px);
  height: calc(100% - var(--editor-header-height));
  display: flex;
  flex-direction: column;
  z-index: 20;
  background-color: var(--theme-body-background);
  overflow-y: hidden;

  /* Using the same colors as the Netmonitor's --table-selection-background-hover */
  --search-result-background-hover: rgba(209, 232, 255, 0.8);
}

.theme-dark .search-container {
  --search-result-background-hover: rgba(53, 59, 72, 1);
}

.project-text-search {
  flex-grow: 1;
  display: flex;
  flex-direction: column;
  overflow-y: hidden;
  height: 100%;
}

.project-text-search .result {
  display: contents;
  cursor: default;
  line-height: 16px;
  font-size: 11px;
  font-family: var(--monospace-font-family);
}

.project-text-search .result:hover > * {
  background-color: var(--search-result-background-hover);
}

.project-text-search .result .line-number {
  grid-column: 1;
  width: 100%;
  padding-block: 1px;
  padding-inline-start: 4px;
  padding-inline-end: 6px;
  text-align: end;
  color: var(--theme-text-color-alt);
}

.project-text-search .result .line-value {
  grid-column: 2;
  padding-block: 1px;
  padding-inline-end: 4px;
  text-overflow: ellipsis;
  overflow-x: hidden;
}

.project-text-search .result .query-match {
  border-bottom: 1px solid var(--theme-contrast-border);
  color: var(--theme-contrast-color);
  background-color: var(--theme-contrast-background);
}

.project-text-search .result.focused .query-match {
  border-bottom: none;
  color: var(--theme-selection-background);
  background-color: var(--theme-selection-color);
}

.project-text-search .tree-indent {
  display: none;
}

.project-text-search .no-result-msg {
  color: var(--theme-text-color-inactive);
  font-size: 24px;
  padding: 4px 15px;
  max-width: 100%;
  overflow-wrap: break-word;
  hyphens: auto;
}

.project-text-search .file-result {
  grid-column: 1/3;
  display: flex;
  align-items: center;
  width: 100%;
  min-height: 24px;
  padding: 2px 4px;
  font-weight: bold;
  font-size: 12px;
  line-height: 16px;
  cursor: default;
}

.project-text-search .file-result .img {
  margin-inline: 2px;
}

.project-text-search .file-result .img.file {
  margin-inline-end: 4px;
}

.project-text-search .file-path {
  flex: 0 1 auto;
  padding-inline-end: 4px;
  display: block;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.project-text-search .file-path:empty {
  display: none;
}

.project-text-search .search-field {
  display: flex;
  align-self: stretch;
  flex-grow: 1;
  width: 100%;
  border-bottom: none;
}

.project-text-search .managed-tree {
  overflow-y: auto;
  height: 100%;
}

.project-text-search .managed-tree .tree {
  height: 100%;
  display: grid;
  /* Align the second column to the search input's text value */
  grid-template-columns: minmax(40px, auto) 1fr;
  overflow-x: hidden;
  padding-top: 4px;
}

/* Fake padding-bottom using a pseudo-element because Gecko doesn't render the
   padding-bottom in a scroll container */
.project-text-search .tree::after {
  content: "";
  display: block;
  height: 4px;
}

.project-text-search .managed-tree .tree .tree-node {
  display: contents;
}

/* Focus values */

.project-text-search .file-result.focused,
.project-text-search .result.focused .line-value,
.project-text-search .result.focused .line-number {
  color: var(--theme-selection-color);
  background-color: var(--theme-selection-background);
}

.project-text-search .file-result.focused .img {
  background-color: currentColor;
}
