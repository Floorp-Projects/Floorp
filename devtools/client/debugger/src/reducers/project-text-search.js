/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

// @format

/**
 * Project text search reducer
 * @module reducers/project-text-search
 */

export const statusType = {
  initial: "INITIAL",
  fetching: "FETCHING",
  cancelled: "CANCELLED",
  done: "DONE",
  error: "ERROR",
};

export function initialProjectTextSearchState() {
  return {
    query: "",
    results: [],
    ongoingSearch: null,
    status: statusType.initial,
  };
}

function update(state = initialProjectTextSearchState(), action) {
  switch (action.type) {
    case "ADD_QUERY":
      return { ...state, query: action.query };

    case "ADD_SEARCH_RESULT":
      if (action.result.matches.length === 0) {
        return state;
      }

      const result = {
        type: "RESULT",
        ...action.result,
        matches: action.result.matches.map(m => ({ type: "MATCH", ...m })),
      };
      return { ...state, results: [...state.results, result] };

    case "UPDATE_STATUS":
      const ongoingSearch =
        action.status == statusType.fetching ? state.ongoingSearch : null;
      return { ...state, status: action.status, ongoingSearch };

    case "CLEAR_SEARCH_RESULTS":
      return { ...state, results: [] };

    case "ADD_ONGOING_SEARCH":
      return { ...state, ongoingSearch: action.ongoingSearch };

    case "CLEAR_SEARCH":
    case "CLOSE_PROJECT_SEARCH":
    case "NAVIGATE":
      return initialProjectTextSearchState();
  }
  return state;
}

export function getTextSearchOperation(state) {
  return state.projectTextSearch.ongoingSearch;
}

export function getTextSearchResults(state) {
  return state.projectTextSearch.results;
}

export function getTextSearchStatus(state) {
  return state.projectTextSearch.status;
}

export function getTextSearchQuery(state) {
  return state.projectTextSearch.query;
}

export default update;
