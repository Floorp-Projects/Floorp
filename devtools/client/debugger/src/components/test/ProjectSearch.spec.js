/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

import React from "react";
import { mount, shallow } from "enzyme";
import { ProjectSearch } from "../ProjectSearch";
import { statusType } from "../../reducers/project-text-search";
import { mockcx } from "../../utils/test-mockup";

const hooks = { on: [], off: [] };
const shortcuts = {
  dispatch(eventName) {
    hooks.on.forEach(hook => {
      if (hook.event === eventName) {
        hook.cb();
      }
    });
    hooks.off.forEach(hook => {
      if (hook.event === eventName) {
        hook.cb();
      }
    });
  },
  on: jest.fn((event, cb) => hooks.on.push({ event, cb })),
  off: jest.fn((event, cb) => hooks.off.push({ event, cb })),
};

const context = { shortcuts };

const testResults = [
  {
    filepath: "testFilePath1",
    type: "RESULT",
    matches: [
      {
        match: "match1",
        value: "some thing match1",
        column: 30,
        type: "MATCH",
      },
      {
        match: "match2",
        value: "some thing match2",
        column: 60,
        type: "MATCH",
      },
      {
        match: "match3",
        value: "some thing match3",
        column: 90,
        type: "MATCH",
      },
    ],
  },
  {
    filepath: "testFilePath2",
    type: "RESULT",
    matches: [
      {
        match: "match4",
        value: "some thing match4",
        column: 80,
        type: "MATCH",
      },
      {
        match: "match5",
        value: "some thing match5",
        column: 40,
        type: "MATCH",
      },
    ],
  },
];

const testMatch = {
  type: "MATCH",
  match: "match1",
  value: "some thing match1",
  sourceId: "some-target/source42",
  line: 3,
  column: 30,
};

function render(overrides = {}, mounted = false) {
  const props = {
    cx: mockcx,
    status: "DONE",
    sources: {},
    results: [],
    query: "foo",
    activeSearch: "project",
    closeProjectSearch: jest.fn(),
    searchSources: jest.fn(),
    clearSearch: jest.fn(),
    updateSearchStatus: jest.fn(),
    selectSpecificLocation: jest.fn(),
    doSearchForHighlight: jest.fn(),
    setActiveSearch: jest.fn(),
    ...overrides,
  };

  return mounted
    ? mount(<ProjectSearch {...props} />, { context })
    : shallow(<ProjectSearch {...props} />, { context });
}

describe("ProjectSearch", () => {
  beforeEach(() => {
    context.shortcuts.on.mockClear();
    context.shortcuts.off.mockClear();
  });

  it("renders nothing when disabled", () => {
    const component = render({ activeSearch: "" });
    expect(component).toMatchSnapshot();
  });

  it("where <Enter> has not been pressed", () => {
    const component = render({ query: "" });
    expect(component).toMatchSnapshot();
  });

  it("found no search results", () => {
    const component = render();
    expect(component).toMatchSnapshot();
  });

  it("should display loading message while search is in progress", () => {
    const component = render({
      query: "match",
      status: statusType.fetching,
    });
    expect(component).toMatchSnapshot();
  });

  it("found search results", () => {
    const component = render(
      {
        query: "match",
        results: testResults,
      },
      true
    );
    expect(component).toMatchSnapshot();
  });

  it("turns off shortcuts on unmount", () => {
    const component = render({
      query: "",
    });
    expect(component).toMatchSnapshot();
    component.unmount();
    expect(context.shortcuts.off).toHaveBeenCalled();
  });

  it("calls inputOnChange", () => {
    const component = render(
      {
        results: testResults,
      },
      true
    );
    component
      .find("SearchInput input")
      .simulate("change", { target: { value: "bar" } });
    expect(component.state().inputValue).toEqual("bar");
  });

  it("onKeyDown Escape/Other", () => {
    const searchSources = jest.fn();
    const component = render(
      {
        results: testResults,
        searchSources,
      },
      true
    );
    component.find("SearchInput input").simulate("keydown", { key: "Escape" });
    expect(searchSources).not.toHaveBeenCalled();
    searchSources.mockClear();
    component
      .find("SearchInput input")
      .simulate("keydown", { key: "Other", stopPropagation: jest.fn() });
    expect(searchSources).not.toHaveBeenCalled();
  });

  it("onKeyDown Enter", () => {
    const searchSources = jest.fn();
    const component = render(
      {
        results: testResults,
        searchSources,
      },
      true
    );
    component
      .find("SearchInput input")
      .simulate("keydown", { key: "Enter", stopPropagation: jest.fn() });
    expect(searchSources).toHaveBeenCalledWith(mockcx, "foo");
  });

  it("onEnterPress shortcut no match or setExpanded", () => {
    const selectSpecificLocation = jest.fn();
    const component = render(
      {
        results: testResults,
        selectSpecificLocation,
      },
      true
    );
    component.instance().state.focusedItem = null;
    shortcuts.dispatch("Enter");
    expect(selectSpecificLocation).not.toHaveBeenCalled();
  });

  it("onEnterPress shortcut match", () => {
    const selectSpecificLocation = jest.fn();
    const component = render(
      {
        results: testResults,
        selectSpecificLocation,
      },
      true
    );
    component.instance().state.focusedItem = { ...testMatch };
    shortcuts.dispatch("Enter");
    expect(selectSpecificLocation).toHaveBeenCalledWith(mockcx, {
      sourceId: "some-target/source42",
      line: 3,
      column: 30,
    });
  });

  it("state.inputValue responds to prop.query changes", () => {
    const component = render({ query: "foo" });
    expect(component.state().inputValue).toEqual("foo");
    component.setProps({ query: "" });
    expect(component.state().inputValue).toEqual("");
  });

  describe("showErrorEmoji", () => {
    it("false if not done & results", () => {
      const component = render({
        status: statusType.fetching,
        results: testResults,
      });
      expect(component).toMatchSnapshot();
    });

    it("false if not done & no results", () => {
      const component = render({
        status: statusType.fetching,
      });
      expect(component).toMatchSnapshot();
    });

    // "false if done & has results"
    // is the same test as "found search results"

    // "true if done & has no results"
    // is the same test as "found no search results"
  });
});
