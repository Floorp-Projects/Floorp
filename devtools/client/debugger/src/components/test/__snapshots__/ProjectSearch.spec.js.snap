// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectSearch found no search results 1`] = `
<div
  className="search-container"
>
  <div
    className="project-text-search"
  >
    <div
      className="header"
    >
      <SearchInput
        count={0}
        expanded={false}
        handleClose={[Function]}
        hasPrefix={false}
        isLoading={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onHistoryScroll={[Function]}
        onKeyDown={[Function]}
        placeholder="Find in files…"
        query="foo"
        selectedItemId=""
        showClose={true}
        showErrorEmoji={true}
        size="big"
        summaryMsg="0 results"
      />
    </div>
    <div
      className="no-result-msg absolute-center"
    >
      No results found
    </div>
  </div>
</div>
`;

exports[`ProjectSearch found search results 1`] = `
<ProjectSearch
  activeSearch="project"
  clearSearch={[MockFunction]}
  closeProjectSearch={[MockFunction]}
  cx={
    Object {
      "navigateCounter": 0,
    }
  }
  doSearchForHighlight={[MockFunction]}
  query="match"
  results={
    Array [
      Object {
        "filepath": "testFilePath1",
        "matches": Array [
          Object {
            "column": 30,
            "match": "match1",
            "type": "MATCH",
            "value": "some thing match1",
          },
          Object {
            "column": 60,
            "match": "match2",
            "type": "MATCH",
            "value": "some thing match2",
          },
          Object {
            "column": 90,
            "match": "match3",
            "type": "MATCH",
            "value": "some thing match3",
          },
        ],
        "type": "RESULT",
      },
      Object {
        "filepath": "testFilePath2",
        "matches": Array [
          Object {
            "column": 80,
            "match": "match4",
            "type": "MATCH",
            "value": "some thing match4",
          },
          Object {
            "column": 40,
            "match": "match5",
            "type": "MATCH",
            "value": "some thing match5",
          },
        ],
        "type": "RESULT",
      },
    ]
  }
  searchSources={[MockFunction]}
  selectSpecificLocation={[MockFunction]}
  setActiveSearch={[MockFunction]}
  sources={Object {}}
  status="DONE"
  updateSearchStatus={[MockFunction]}
>
  <div
    className="search-container"
  >
    <div
      className="project-text-search"
    >
      <div
        className="header"
      >
        <SearchInput
          count={5}
          expanded={false}
          handleClose={[Function]}
          hasPrefix={false}
          isLoading={false}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onHistoryScroll={[Function]}
          onKeyDown={[Function]}
          placeholder="Find in files…"
          query="match"
          selectedItemId=""
          showClose={true}
          showErrorEmoji={false}
          size="big"
          summaryMsg="5 results"
        >
          <div
            className="search-outline"
          >
            <div
              aria-expanded={false}
              aria-haspopup="listbox"
              aria-owns="result-list"
              className="search-field big"
              role="combobox"
            >
              <AccessibleImage
                className="search"
              >
                <span
                  className="img search"
                />
              </AccessibleImage>
              <input
                aria-activedescendant=""
                aria-autocomplete="list"
                aria-controls="result-list"
                className=""
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="Find in files…"
                spellCheck={false}
                value="match"
              />
              <div
                className="search-field-summary"
              >
                5 results
              </div>
              <CloseButton
                buttonClass="big"
                handleClick={[Function]}
              >
                <button
                  className="close-btn big"
                  onClick={[Function]}
                >
                  <AccessibleImage
                    className="close"
                  >
                    <span
                      className="img close"
                    />
                  </AccessibleImage>
                </button>
              </CloseButton>
            </div>
          </div>
        </SearchInput>
      </div>
      <ManagedTree
        autoExpandAll={true}
        autoExpandDepth={1}
        autoExpandNodeChildrenLimit={100}
        focused={null}
        getChildren={[Function]}
        getParent={[Function]}
        getPath={[Function]}
        getRoots={[Function]}
        itemHeight={24}
        onFocus={[Function]}
        renderItem={[Function]}
      >
        <div
          className="managed-tree"
        >
          <Tree
            autoExpandAll={true}
            autoExpandDepth={1}
            autoExpandNodeChildrenLimit={100}
            focused={null}
            getChildren={[Function]}
            getKey={[Function]}
            getParent={[Function]}
            getPath={[Function]}
            getRoots={[Function]}
            isExpanded={[Function]}
            itemHeight={24}
            onCollapse={[Function]}
            onExpand={[Function]}
            onFocus={[Function]}
            renderItem={[Function]}
          >
            <div
              aria-activedescendant={null}
              className="tree "
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              role="tree"
              style={Object {}}
              tabIndex="0"
            >
              <TreeNode
                active={false}
                depth={0}
                expanded={true}
                focused={false}
                id="undefined-$"
                index={0}
                isExpandable={true}
                item={
                  Object {
                    "filepath": "testFilePath1",
                    "matches": Array [
                      Object {
                        "column": 30,
                        "match": "match1",
                        "type": "MATCH",
                        "value": "some thing match1",
                      },
                      Object {
                        "column": 60,
                        "match": "match2",
                        "type": "MATCH",
                        "value": "some thing match2",
                      },
                      Object {
                        "column": 90,
                        "match": "match3",
                        "type": "MATCH",
                        "value": "some thing match3",
                      },
                    ],
                    "type": "RESULT",
                  }
                }
                key="undefined-$-inactive"
                onClick={[Function]}
                onCollapse={[Function]}
                onExpand={[Function]}
                renderItem={[Function]}
              >
                <div
                  aria-expanded={true}
                  aria-level={1}
                  className="tree-node"
                  data-expandable={true}
                  id="undefined-$"
                  onClick={[Function]}
                  onKeyDownCapture={null}
                  role="treeitem"
                >
                  <div
                    className="file-result"
                  >
                    <AccessibleImage
                      className="arrow expanded"
                    >
                      <span
                        className="img arrow expanded"
                      />
                    </AccessibleImage>
                    <AccessibleImage
                      className="file"
                    >
                      <span
                        className="img file"
                      />
                    </AccessibleImage>
                    <span
                      className="file-path"
                    />
                    <span
                      className="matches-summary"
                    >
                       (3 matches)
                    </span>
                  </div>
                </div>
              </TreeNode>
              <TreeNode
                active={false}
                depth={1}
                expanded={false}
                focused={false}
                id="undefined-undefined-30-1"
                index={1}
                isExpandable={false}
                item={
                  Object {
                    "column": 30,
                    "match": "match1",
                    "type": "MATCH",
                    "value": "some thing match1",
                  }
                }
                key="undefined-undefined-30-1-inactive"
                onClick={[Function]}
                onCollapse={[Function]}
                onExpand={[Function]}
                renderItem={[Function]}
              >
                <div
                  aria-level={2}
                  className="tree-node"
                  data-expandable={false}
                  id="undefined-undefined-30-1"
                  onClick={[Function]}
                  onKeyDownCapture={null}
                  role="treeitem"
                >
                  <span
                    className="tree-indent tree-last-indent"
                  >
                    ​
                  </span>
                  <div
                    className="result"
                    onClick={[Function]}
                  >
                    <span
                      className="line-number"
                    />
                    <span
                      className="line-value"
                    >
                      <span
                        className="line-match"
                        key="0"
                      >
                        some thing match1
                      </span>
                      <span
                        className="query-match"
                        key="1"
                      >
                        some t
                      </span>
                      <span
                        className="line-match"
                        key="2"
                      >
                        some thing match1
                      </span>
                    </span>
                  </div>
                </div>
              </TreeNode>
              <TreeNode
                active={false}
                depth={1}
                expanded={false}
                focused={false}
                id="undefined-undefined-60-2"
                index={2}
                isExpandable={false}
                item={
                  Object {
                    "column": 60,
                    "match": "match2",
                    "type": "MATCH",
                    "value": "some thing match2",
                  }
                }
                key="undefined-undefined-60-2-inactive"
                onClick={[Function]}
                onCollapse={[Function]}
                onExpand={[Function]}
                renderItem={[Function]}
              >
                <div
                  aria-level={2}
                  className="tree-node"
                  data-expandable={false}
                  id="undefined-undefined-60-2"
                  onClick={[Function]}
                  onKeyDownCapture={null}
                  role="treeitem"
                >
                  <span
                    className="tree-indent tree-last-indent"
                  >
                    ​
                  </span>
                  <div
                    className="result"
                    onClick={[Function]}
                  >
                    <span
                      className="line-number"
                    />
                    <span
                      className="line-value"
                    >
                      <span
                        className="line-match"
                        key="0"
                      >
                        some thing match2
                      </span>
                      <span
                        className="query-match"
                        key="1"
                      >
                        some t
                      </span>
                      <span
                        className="line-match"
                        key="2"
                      >
                        some thing match2
                      </span>
                    </span>
                  </div>
                </div>
              </TreeNode>
              <TreeNode
                active={false}
                depth={1}
                expanded={false}
                focused={false}
                id="undefined-undefined-90-3"
                index={3}
                isExpandable={false}
                item={
                  Object {
                    "column": 90,
                    "match": "match3",
                    "type": "MATCH",
                    "value": "some thing match3",
                  }
                }
                key="undefined-undefined-90-3-inactive"
                onClick={[Function]}
                onCollapse={[Function]}
                onExpand={[Function]}
                renderItem={[Function]}
              >
                <div
                  aria-level={2}
                  className="tree-node"
                  data-expandable={false}
                  id="undefined-undefined-90-3"
                  onClick={[Function]}
                  onKeyDownCapture={null}
                  role="treeitem"
                >
                  <span
                    className="tree-indent tree-last-indent"
                  >
                    ​
                  </span>
                  <div
                    className="result"
                    onClick={[Function]}
                  >
                    <span
                      className="line-number"
                    />
                    <span
                      className="line-value"
                    >
                      <span
                        className="line-match"
                        key="0"
                      >
                        some thing match3
                      </span>
                      <span
                        className="query-match"
                        key="1"
                      >
                        some t
                      </span>
                      <span
                        className="line-match"
                        key="2"
                      >
                        some thing match3
                      </span>
                    </span>
                  </div>
                </div>
              </TreeNode>
              <TreeNode
                active={false}
                depth={0}
                expanded={true}
                focused={false}
                id="undefined-4"
                index={4}
                isExpandable={true}
                item={
                  Object {
                    "filepath": "testFilePath2",
                    "matches": Array [
                      Object {
                        "column": 80,
                        "match": "match4",
                        "type": "MATCH",
                        "value": "some thing match4",
                      },
                      Object {
                        "column": 40,
                        "match": "match5",
                        "type": "MATCH",
                        "value": "some thing match5",
                      },
                    ],
                    "type": "RESULT",
                  }
                }
                key="undefined-4-inactive"
                onClick={[Function]}
                onCollapse={[Function]}
                onExpand={[Function]}
                renderItem={[Function]}
              >
                <div
                  aria-expanded={true}
                  aria-level={1}
                  className="tree-node"
                  data-expandable={true}
                  id="undefined-4"
                  onClick={[Function]}
                  onKeyDownCapture={null}
                  role="treeitem"
                >
                  <div
                    className="file-result"
                  >
                    <AccessibleImage
                      className="arrow expanded"
                    >
                      <span
                        className="img arrow expanded"
                      />
                    </AccessibleImage>
                    <AccessibleImage
                      className="file"
                    >
                      <span
                        className="img file"
                      />
                    </AccessibleImage>
                    <span
                      className="file-path"
                    />
                    <span
                      className="matches-summary"
                    >
                       (2 matches)
                    </span>
                  </div>
                </div>
              </TreeNode>
              <TreeNode
                active={false}
                depth={1}
                expanded={false}
                focused={false}
                id="undefined-undefined-80-5"
                index={5}
                isExpandable={false}
                item={
                  Object {
                    "column": 80,
                    "match": "match4",
                    "type": "MATCH",
                    "value": "some thing match4",
                  }
                }
                key="undefined-undefined-80-5-inactive"
                onClick={[Function]}
                onCollapse={[Function]}
                onExpand={[Function]}
                renderItem={[Function]}
              >
                <div
                  aria-level={2}
                  className="tree-node"
                  data-expandable={false}
                  id="undefined-undefined-80-5"
                  onClick={[Function]}
                  onKeyDownCapture={null}
                  role="treeitem"
                >
                  <span
                    className="tree-indent tree-last-indent"
                  >
                    ​
                  </span>
                  <div
                    className="result"
                    onClick={[Function]}
                  >
                    <span
                      className="line-number"
                    />
                    <span
                      className="line-value"
                    >
                      <span
                        className="line-match"
                        key="0"
                      >
                        some thing match4
                      </span>
                      <span
                        className="query-match"
                        key="1"
                      >
                        some t
                      </span>
                      <span
                        className="line-match"
                        key="2"
                      >
                        some thing match4
                      </span>
                    </span>
                  </div>
                </div>
              </TreeNode>
              <TreeNode
                active={false}
                depth={1}
                expanded={false}
                focused={false}
                id="undefined-undefined-40-6"
                index={6}
                isExpandable={false}
                item={
                  Object {
                    "column": 40,
                    "match": "match5",
                    "type": "MATCH",
                    "value": "some thing match5",
                  }
                }
                key="undefined-undefined-40-6-inactive"
                onClick={[Function]}
                onCollapse={[Function]}
                onExpand={[Function]}
                renderItem={[Function]}
              >
                <div
                  aria-level={2}
                  className="tree-node"
                  data-expandable={false}
                  id="undefined-undefined-40-6"
                  onClick={[Function]}
                  onKeyDownCapture={null}
                  role="treeitem"
                >
                  <span
                    className="tree-indent tree-last-indent"
                  >
                    ​
                  </span>
                  <div
                    className="result"
                    onClick={[Function]}
                  >
                    <span
                      className="line-number"
                    />
                    <span
                      className="line-value"
                    >
                      <span
                        className="line-match"
                        key="0"
                      >
                        some thing match5
                      </span>
                      <span
                        className="query-match"
                        key="1"
                      >
                        some t
                      </span>
                      <span
                        className="line-match"
                        key="2"
                      >
                        some thing match5
                      </span>
                    </span>
                  </div>
                </div>
              </TreeNode>
            </div>
          </Tree>
        </div>
      </ManagedTree>
    </div>
  </div>
</ProjectSearch>
`;

exports[`ProjectSearch renders nothing when disabled 1`] = `""`;

exports[`ProjectSearch should display loading message while search is in progress 1`] = `
<div
  className="search-container"
>
  <div
    className="project-text-search"
  >
    <div
      className="header"
    >
      <SearchInput
        count={0}
        expanded={false}
        handleClose={[Function]}
        hasPrefix={false}
        isLoading={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onHistoryScroll={[Function]}
        onKeyDown={[Function]}
        placeholder="Find in files…"
        query="match"
        selectedItemId=""
        showClose={true}
        showErrorEmoji={false}
        size="big"
        summaryMsg="0 results"
      />
    </div>
    <div
      className="no-result-msg absolute-center"
    >
      Loading…
    </div>
  </div>
</div>
`;

exports[`ProjectSearch showErrorEmoji false if not done & no results 1`] = `
<div
  className="search-container"
>
  <div
    className="project-text-search"
  >
    <div
      className="header"
    >
      <SearchInput
        count={0}
        expanded={false}
        handleClose={[Function]}
        hasPrefix={false}
        isLoading={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onHistoryScroll={[Function]}
        onKeyDown={[Function]}
        placeholder="Find in files…"
        query="foo"
        selectedItemId=""
        showClose={true}
        showErrorEmoji={false}
        size="big"
        summaryMsg="0 results"
      />
    </div>
    <div
      className="no-result-msg absolute-center"
    >
      Loading…
    </div>
  </div>
</div>
`;

exports[`ProjectSearch showErrorEmoji false if not done & results 1`] = `
<div
  className="search-container"
>
  <div
    className="project-text-search"
  >
    <div
      className="header"
    >
      <SearchInput
        count={5}
        expanded={false}
        handleClose={[Function]}
        hasPrefix={false}
        isLoading={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onHistoryScroll={[Function]}
        onKeyDown={[Function]}
        placeholder="Find in files…"
        query="foo"
        selectedItemId=""
        showClose={true}
        showErrorEmoji={false}
        size="big"
        summaryMsg="5 results"
      />
    </div>
    <ManagedTree
      autoExpandAll={true}
      autoExpandDepth={1}
      autoExpandNodeChildrenLimit={100}
      focused={null}
      getChildren={[Function]}
      getParent={[Function]}
      getPath={[Function]}
      getRoots={[Function]}
      itemHeight={24}
      onFocus={[Function]}
      renderItem={[Function]}
    />
  </div>
</div>
`;

exports[`ProjectSearch turns off shortcuts on unmount 1`] = `
<div
  className="search-container"
>
  <div
    className="project-text-search"
  >
    <div
      className="header"
    >
      <SearchInput
        count={0}
        expanded={false}
        handleClose={[Function]}
        hasPrefix={false}
        isLoading={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onHistoryScroll={[Function]}
        onKeyDown={[Function]}
        placeholder="Find in files…"
        query=""
        selectedItemId=""
        showClose={true}
        showErrorEmoji={true}
        size="big"
        summaryMsg=""
      />
    </div>
  </div>
</div>
`;

exports[`ProjectSearch where <Enter> has not been pressed 1`] = `
<div
  className="search-container"
>
  <div
    className="project-text-search"
  >
    <div
      className="header"
    >
      <SearchInput
        count={0}
        expanded={false}
        handleClose={[Function]}
        hasPrefix={false}
        isLoading={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onHistoryScroll={[Function]}
        onKeyDown={[Function]}
        placeholder="Find in files…"
        query=""
        selectedItemId=""
        showClose={true}
        showErrorEmoji={true}
        size="big"
        summaryMsg=""
      />
    </div>
  </div>
</div>
`;
